Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculation.kt

```diff
@@ -31,10 +34,11 @@ class DccWalletInfoCalculation @Inject constructor(
         boosterRulesNode = gson.toJson(boosterRules).toJsonNode()
     }
 
-    fun getDccWalletInfo(
+    @Suppress("BlockingMethodInNonBlockingContext")
+    suspend fun getDccWalletInfo(
         dccList: List<CwaCovidCertificate>,
         dateTime: DateTime = DateTime.now()
-    ): DccWalletInfo {
+    ): DccWalletInfo = withContext(dispatcherProvider.IO) {
```

Comment:
dispatcherProvider.Default should be used in this case I think because it's not really IO related work that is performed but CPU heavy work. However, the Dispatcher switch should happen in CclJsonFunctions.evaluateFunctions() so that not all callers of this functions need to switch to the default dispatcher before.

Reply:
1- if you dig deeper in `writeTree()` you will see that it is actually doing `IO` operations with streams
2- I did not get what you mean by `the Dispatcher switch should happen within CclJsonFunctions.evaluateFunction() so that not all callers of this functions need to switch to the default dispatcher before.`
I think we should switch the dispatcher in a common place where you know actually what context you need. the caller of this function is not aware of that

Reply:
1 - Ah ok, I see - yes then the mapper can work on the IO Dispatcher. However the evaluation with JsonFunctions should happen on the Default Dispatcher IMO. 

2 - I mean that within `CCLJsonFunction` the evaluateFunction looks like this: 

```kotlin

suspend fun evaluateFunction(
        functionName: String,
        parameters: JsonNode
    ) = withContext(dispatcherProvider.Default) {
        ...
    }

```

This way, `DccWalletInfoCalculation` doesn't need to perform the dispatcher switch. Also new Classes that call `evaluateFunction()` (like the system-time-dependent time formatter) don't need to perform that switch.

Reply:
done